### ISSUE TITLE
all tests should run on each PR, and alert us if they fail.

### ISSUE THREAD

**Issue by holmesworcester** [TryQuiet/quiet#587]
Right now e2e tests can fail and we won't notice it until the next build. This wastes a lot of the power of e2e tests.

We should get an alert if e2e tests fail (even if it's several hours after the merge that caused the failure) and we should be able to see a list of PRs and quickly identify which one caused the failure.

We should also run tests again on release.

If tests are too slow to block merging, we should decide how much wait is acceptable, run as many useful tests as we can in that timeframe, and then run all other tests in a way that does not block merging but will still inform us of failures.

Or we should find a way to speed up our tests.

See #588 for how we could go farther with this.

**Comment by holmesworcester** [comment]
If fixing e2e tests is impractical, we should at least set up a test to ensure that Quiet builds correctly for all platforms, and run this daily.

**Comment by Kacper-RF** [comment]
Link to related issue - Automated approach for E2E tests - https://github.com/TryQuiet/quiet/issues/1334

**Comment by Kacper-RF** [comment]
-Versions for tests are now builded dynamically on every workflow trigger, so they included newest changes(previously they were downloaded from github releases, from newest release)
-Tests are triggered on every pull request
-They can take more then 30min, but they don't block merge
-After fail in tests commit creator will receive mail about failed workflow

What now I think it will be worthy to do is triggering the workflow before every release

**Comment by holmesworcester** [comment]
Can we trigger the workflow manually before releases? If so let's create a separate issue for this so we can move on.

**Comment by holmesworcester** [comment]
@Kacper-RF are these running before release now?

**Comment by Kacper-RF** [comment]
@holmesworcester 
Yes, E2E test run before release, we only have little problem with one step for Windows.
I will assume that step with Windows certificate can be triggered only once at the whole workflow, and now it's triggered two times(because we build version for tests and release),it would need to be checked

macOS and Linux on two versions are working very great, after tests passed they triggering release job

**Comment by holmesworcester** [comment]
why only once?

**Comment by Kacper-RF** [comment]
It's just my thought why this step might crashing, because build-release CI it's only one which use that step two times.
It will take some time to check this
